backend/get_version.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
```

Comment:
This is executed at build time (e.g. at image build time), and can be overridden with `UV_DYNAMIC_VERSIONING_BYPASS` (which is hence added as an `ARG` to the Dockerfile).

Reply:
Yep, I think we should indeed do this. I somewhat postponed addressing this though by not actually making the Docker images posted to the OCI registry in this PR yet.